app/controllers/users.js

```diff
@@ -54,3 +56,26 @@ exports.signup = (req, res, next) => {
     next(err);
   });
 };
+
+exports.signin = (req, res, next) => {
+  const user = req.body ? req.body : {};
+  emailValidation(req.body.email);
+  userService.getByEmail(user.email).then((foundUser) => {
+    if (foundUser) {
+      bcrypt.compare(user.password, foundUser.password).then((match) => {
+        if (match) {
+          const token = sessionManager.encode({ username: foundUser.email });
+          res.status(200);
+          res.set(sessionManager.HEADER_NAME, token);
+          res.send(foundUser);
+        } else {
+          next(errors.defaultError('The password is incorrect'));
```

Comment:
We should use a different error, remenber that the defaultError means a 500 status code, which means that something has gone wrong unexpectedly in the server. But here the error is that the password provided by the user is invalid, so the error code should be some of the 4xx family.